### PR TITLE
(retriever) Add .store() task for persisting extracted images (#1675)

### LINKED ISSUE #1675: [FEA]: nemo_retriever library: Support storing extracted images to disk
### Is this a new feature, an improvement, or a change to existing functionality?

New Feature

### How would you describe the priority of this feature request

Currently preventing usage

### Please provide a clear description of problem this feature solves

Like the old nv_ingest API, nemo_retriever library needs a `.store()` task which supports saving to disk:
1. full page images
2. sub-page images (cropped images of recognized tables, charts, infographics, etc)

### Describe the feature, and optionally a solution or implementation and any alternatives

It should use the fsspec package so that users can have these images persisted to cloud/other distributed storage if desired.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jioffe502**
## Summary

Adds a new `.store()` pipeline task to `nemo_retriever` that persists extracted images (full-page + sub-page table/chart/infographic/image assets) to local or cloud storage via fsspec/UPath.

- Wire `.store()` into `InProcessIngestor`, `BatchIngestor` (`map_batches`), and batch CLI via `--store-images-uri`
- Add `StoreParams` with per-content toggles, `image_format`, and `strip_base64` (default `False` — opt-in memory reduction, preserves multimodal embedding compatibility)
- Ensure direct-write file extensions match actual encoded bytes (declared `encoding` field + magic-byte sniff fallback); cropped outputs follow configured `image_format`
- Fine-grained control (per-content-type toggles, `public_base_url`, `storage_options`) available via `StoreParams` in library use; CLI exposes `--store-images-uri` for common case

## Test plan

- `test_io_image_store.py`: page images, structured crops, natural images, format consistency, stripping, edge cases
- `test_multimodal_embed.py`: verifies store → embed behavior for multimodal with/without stripping
- Integration: `bo20` dataset (20 PDFs, 496 pages) → 842 stored assets (496 page images + 346 table/chart crops)

```bash
uv run pytest tests/test_io_image_store.py tests/test_multimodal_embed.py -q
```

## Known limitations (follow-ups)

- **Schema coupling:** store targets singular structured columns (`table`, `chart`, `infographic`) matching OCR stage output
- **Corrupt `image_b64` skip:** if direct `image_b64` exists but decode fails, we skip rather than falling back to `bbox_xyxy_norm` crop from page image
- **`_safe_stem` collision:** two files with same basename but different directories share output subdirectory; hash-based naming is a follow-up
- **Observability:** per-type write counters (pages, tables, charts, failures) deferred

Closes #1675

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: nemo_retriever/src/nemo_retriever/params/models.py [hunk 1]
```diff
@@ -270,6 +270,19 @@ class VdbUploadParams(_ParamsModel):
     lancedb: LanceDbParams = Field(default_factory=LanceDbParams)
 
 
+class StoreParams(_ParamsModel):
```
Comment (jperez999) [inline nemo_retriever/src/nemo_retriever/params/models.py:273]: Does this allow me to store full chunks also... I think that is the only permutation you dont have here, text..?

File: nemo_retriever/tests/test_io_image_store.py [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+import io
+from pathlib import Path
+
+import pandas as pd
+
+from nemo_retriever.io.image_store import _safe_stem, load_image_b64_from_uri, store_extracted_images
+from nemo_retriever.params import StoreParams
+
+
+def _make_tiny_png_b64(width: int = 4, height: int = 4, color=(255, 0, 0)) -> str:
```
Comment (jperez999) [inline nemo_retriever/tests/test_io_image_store.py:19]: Some of these functions are doubled up in other files. We are going to need to consolidate that.
